src/robot/libraries/XML.py

```diff
@@ -821,11 +869,11 @@ def get_element_attributes(self, source, xpath="."):
 
     def element_attribute_should_be(
         self,
-        source,
-        name,
-        expected,
-        xpath=".",
-        message=None,
+        source: ElementOrUnparsed,
+        name: "bytes | str",
+        expected: "None | bytes | str",
```

Comment:
`bytes` most likely isn't needed and is actually wrong. See the previous comments for more info.

I'd prefer `expected` typing to be `str | None` instead of `None | str` because `str` is the more common type. The order shouldn't affect argument conversion in this case.